flow/storage/elasticsearch/elasticsearch.go

```diff
@@ -161,13 +161,16 @@ const rawPacketMapping = `
 // ElasticSearchStorage describes an ElasticSearch flow backend
 type ElasticSearchStorage struct {
 	client *esclient.ElasticSearchClient
+	perf   map[string]*common.PerfCounterIntRate
 }
 
 // StoreFlows push a set of flows in the database
 func (c *ElasticSearchStorage) StoreFlows(flows []*flow.Flow) error {
 	if !c.client.Started() {
 		return errors.New("ElasticSearchStorage is not yet started")
 	}
+	//@perf c.perf["StoreFlows"].Prolog(int64(len(flows)))
```

Comment:
I would prefer to add one tag/comment before the function definition, your perl script would read this tag and would inject these two lines at the beginning of the function using the name of the function+the path. I think it is too intrusive to add things in the middle of the functions. if we need to provide more fine grained stuff to the users then we should split functions but that is another story.

Reply:
I think tagging the line give us flexibility that we want to inject the code that, specially in the case of inside loop (see flow/table.go)
In the case of StoreFlows for example, we didn't want to count StoreFlows call/timing when the client is not connected

Reply:
Let's start with something simple and non intrusive, here it is really too intrusive. If we want to get fine grained metrics let us the profiler otherwise something simple will be fine and if at some point we need more flexibility we will improve it

Reply:
keep in mind this patch is to measure internal performances during runtime more like a monitoring tool ... profiling we let known already on calls and stacktrace of functions
Where do you feel this is intrusive ? in all case the code will not run on production/normal build/release